src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -55,11 +56,20 @@ export async function getSharingData(
 
   return {
     getSearchSource: (absoluteTime?: boolean): SearchSourceFields => {
-      const filter = absoluteTime
+      const timeFilter = absoluteTime
         ? data.query.timefilter.timefilter.createFilter(index)
         : data.query.timefilter.timefilter.createRelativeFilter(index);
 
-      searchSource.setField('filter', filter);
+      if (existingFilter && timeFilter) {
+        searchSource.setField(
```

Comment:
I wonder how common this append to filter approach is. Perhaps it should be part of search source @elastic/kibana-app-services ?

Reply:
cc @ppisljar